src/sentry/integrations/utils/metrics.py

```diff
@@ -197,3 +203,48 @@ def __exit__(
                 if self.assume_success
                 else EventLifecycleOutcome.HALTED
             )
+
+
+class IntegrationPipelineViewType(Enum):
+    """A specific step in an integration's pipeline that is not a static page."""
+
+    # IdentityProviderPipeline
+    IDENTITY_PROVIDER = "IDENTITY_PROVIDER"
+
+    # GitHub
+    OAUTH_LOGIN = "OAUTH_LOGIN"
+    GITHUB_INSTALLATION = "GITHUB_INSTALLATION"
+
+    # Bitbucket
+    VERIFY_INSTALLATION = "VERIFY_INSTALLATION"
+
+    # Bitbucket Server
+    # OAUTH_LOGIN = "OAUTH_LOGIN"
+    OAUTH_CALLBACK = "OAUTH_CALLBACK"
+
+    # Azure DevOps
+    ACCOUNT_CONFIG = "ACCOUNT_CONFIG"
+
+    def __str__(self) -> str:
+        return self.value.lower()
+
+
+@dataclass
+class IntegrationPipelineViewEvent(EventLifecycleMetric):
+    """An instance to be recorded of a user going through an integration pipeline view (step)."""
+
+    interaction_type: IntegrationPipelineViewType
+    domain: str
+    provider_key: str
+
+    def get_key(self, outcome: EventLifecycleOutcome) -> str:
+        # not reporting as SLOs
+        root_tokens = ("sentry", "integrations", "installation")
+        specific_tokens = (
+            self.domain,
+            self.provider_key,
+            str(self.interaction_type),
+            str(outcome),
+        )
```

Comment:
I'm open to suggestions if the `get_standard_key` method could be made more flexible. In a perfect world, we wouldn't have to duplicate the `".".join(...)` stuff. This is perfectly fine for now though; it's easy to refactor later.

The "slo" token was mostly just a catch-all to distinguish the new metrics. On further reflection, I'm not interested in codifying a separation between metrics that are targeted at establishing SLOs versus ones that aren't. (The code's only responsibility is to make sure the metrics are clear and accurate. Whether they inform SLOs or not is a concern at another layer IMHO.) So I'll probably be changing that.